### ISSUE TITLE
Error: rfnocmodtool always asks for src newmod directory in a Custom Prefix Installation (without pybombs) Environment

### ISSUE THREAD

**Issue by yaaseen-dev** [EttusResearch/gr-ettus#45]
I have installed gr-ettus, gnuradio, uhd in a custom prefix 
~/workspace/installs/stable/

I was trying to build a custom RFNoC module using rfnocmodtool. So when I did I am getting the following error :  

```
ya-seen@sdr-linux-machine:~$ source workspace/installs/stable/setup.env 
ya-seen@sdr-linux-machine:~$ rfnocmodtool newmod MyFirstnoc
Could not find rfnoc-newmod source dir. 
Please run: 

	 $ rfnocmodtool newmod [NAME] --srcdir {path/to/rfnoc-newmod/}

Specifying the path where the newmod template is located.
```

even after supplying a --srcdir param I was getting the same error

```
 ya-seen@sdr-linux-machine:~$ rfnocmodtool newmod MyFirstnoc --srcdir /workspace/installs/stable/share/gr-ettus/rfnoc_modtool/rfnoc-newmod
Could not find rfnoc-newmod source dir. 
Please run: 

	 $ rfnocmodtool newmod [NAME] --srcdir {path/to/rfnoc-newmod/}

Specifying the path where the newmod template is located.
```

**Comment by yaaseen-dev** [comment]
I did few debugging myself. I found out that, the code which does this is in the file:
workspace/installs/stable/lib/python3/dist-packages/ettus/rfnoc_modtool/modtool_newmod.py

particularly this code 

```
 self._srcdir = gr.prefs().get_string('rfnocmodtool', 'newmod_path',
                                             args.srcdir)
        if not os.path.isdir(self._srcdir):
            raise ModToolException("Could not find rfnoc-newmod source dir. \n"
                    "Please run: \n\n\t $ rfnocmodtool newmod [NAME] --srcdir"
                    " {path/to/rfnoc-newmod/}\n\n"
                    "Specifying the path where the newmod template is located.")
```

```
gr.prefs().get_string('rfnocmodtool', 'newmod_path',
                                             args.srcdir)
```
This is the exact line that gets the path from the  _rfnocmodtool.conf_ file. In this it has this path:

`newmod_path = /home/ya-seen/workspace/installs/stable/share/gr-ettus/rfnoc_modtool/python/rfnoc-newmod`

But in the actual the rfnoc-newmod folder gets installed at  `{USER_PREFIX}/share/gr-ettus/rfnoc_modtool/rfnoc-newmod`

Hence the rfnocmodtool command throws error

**Comment by yaaseen-dev** [comment]
You can confirm this from the cmake file present at `gr-ettus/python/rfnoc_modtool/CMakeLists.txt `

```
set(GR_PKG_RFNOC_MODTOOL_DATA_DIR ${GR_PKG_DATA_DIR}/rfnoc_modtool)
install(DIRECTORY rfnoc-newmod
    DESTINATION ${GR_PKG_RFNOC_MODTOOL_DATA_DIR}
    COMPONENT "utils"
    )

#################################################################
# Create and install the rfnoc_modtool conf file
#################################################################
file(TO_NATIVE_PATH "${CMAKE_INSTALL_PREFIX}/${GR_PKG_RFNOC_MODTOOL_DATA_DIR}/python/rfnoc-newmod" newmoddir)
```
Here you can see is installed to GR_PKG_RFNOC_MODTOOL_DATA_DIR =  ${GR_PKG_DATA_DIR}/rfnoc_modtool)

where GR_PKG_DATA_DIR = share/gr-ettus

But the .conf is set to ${CMAKE_INSTALL_PREFIX}/share/gr-ettus/rfnoc_modtool/python/rfnoc-newmod"

**Comment by PE1NUT** [comment]
Running into a similar problem.

I built gr-ettus with -DCMAKE_INSTALL_PREFIX=/usr

Contents of /etc/gnuradio.conf.d/rfnocmodtool.conf:
newmod_path = /usr/share/gr-ettus/rfnoc_modtool/python/rfnoc-newmod

However, the above path is incorrect, as the rfnoc-newmod gets installed in
/usr/share/gr-ettus/rfnoc_modtool/rfnoc-newmod

Fixing the configuration file makes no difference, because (using strace) it turns out that rfnocmodtool still looks in /usr/local:

stat("/usr/local/share/gr-ettus/rfnoc_modtool/rfnoc-newmod", 0x7ffd79109970) = -1 ENOENT (No such file or directory)
write(2, "Could not find rfnoc-newmod sour"..., 178Could not find rfnoc-newmod source dir.

**Comment by bpadalino** [comment]
I have added a last resort search for a relative path to the python script being run in pull request #47.  I hope this is helpful.

**Comment by ArtisticZhao** [comment]
I had use the arg `--srcdir <uhd-install-prefix>/share/gr-ettus/rfnoc_modtool/rfnoc-newmod` to fix that.

**Comment by mbr0wn** [comment]
I merged #47 in the hope this fixes this -- haven't tested. Please reopen if issue persists to be a problem.
